DataConnectors/CEF/cef_gather_info.py

```diff
@@ -0,0 +1,166 @@
+#       CentOS 7 and 8
+#       Oracle Linux 7
+#       Red Hat Enterprise Linux Server 7 and 8
+#       Debian GNU/Linux 8 and 9
+#       Ubuntu Linux 14.04 LTS and 16.04 LTS
+# For more information please check the OMS-Agent-for-Linux documentation.
+#
+
+import time
+import subprocess
+
+
+class SystemInfo:
+    def __init__(self, command, command_output="No output received"):
+        self.command = command
+        self.command_output = command_output
+
+    def __repr__(self):
+        delimiter = "\n-----------------------------------\n"
+        return str(
+            delimiter + "command: " + self.command + '\n' + "output: " + self.command_output + delimiter)
+
+
+script_version = '1.0'
+output_file_path = '/tmp/cef_get_info'
+
+basic_command_dict = {
+    "script_version": "echo {}".format(script_version),
+    "date": "sudo date",
+    "netstat": "sudo netstat -lnpvt",
+    "df": "sudo df -h",
+    "free": "sudo free -m",
+    "iptables": "sudo iptables -vnL --line",
+    "selinux": "sudo cat /etc/selinux/config",
+    "os_version": "sudo cat /etc/issue",
+    "python_version": "sudo python -V",
+    "ram_stats": "sudo cat /proc/meminfo",
+    "cron_jobs": "sudo crontab -l",
+    "wd_list": "sudo ls -l .",
+    "internet_connection": "sudo curl -D - http://google.com",
+    "sudoers_list": "sudo cat /etc/sudoers",
+    "rotation_configuration": "sudo cat /etc/logrotate.conf",
+    "rsyslog_conf": "sudo cat /etc/rsyslog.conf",
+    "rsyslog_dir": "sudo ls -l /etc/rsyslog.d/",
+    "rsyslog_dir_content": "sudo find /etc/rsyslog.d/ -type f -exec cat {} \;",
+    "syslog_ng_conf": "sudo cat /etc/syslog-ng/syslog-ng.conf",
+    "syslog_ng_dir": "sudo ls -l /etc/syslog-ng/conf.d/",
+    "syslog_ng_dir_content": "find /etc/syslog-ng/conf.d/ -type f -exec cat {} \;",
+    "agent_log_snip": "sudo tail -15 /var/opt/microsoft/omsagent/log/omsagent.log",
+    "agent_config_dir": "sudo ls -lR /etc/opt/microsoft/omsagent/",
+    "agent_cef_config": "sudo cat /etc/opt/microsoft/omsagent/conf/omsagent.d/security_events.conf",
+    "tcpdump": "sudo timeout 2 tcpdump -A -ni any port 25226 -vv"
+}
+
+# command with a pipe are considered special commands
+advanced_command_dict = {
+    "top_processes": ["sudo top -bcn1 -w512", "head -n 20"],
+    "omsagent_process": ["sudo ps -aux", "grep omsagent"]
+}
+
+
+def print_notice(input_str):
+    '''
+    Print given text in white background
+    :param input_str:
+    '''
+    print("\033[0;30;47m" + input_str + "\033[0m")
+
+
+def append_content_to_file(command_object, file_path=output_file_path):
+    """
+
+    :param command_object: consists of the name and the output
+    :param file_path: a file to share the commands outputs
+    """
+    output = repr(command_object).replace('%', '%%')
+    command_tokens = ["sudo", "bash", "-c", "printf '" + "\n" + output + "' >> " + file_path]
+    try:
+        write_new_content = subprocess.Popen(command_tokens, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
```

Comment:
subprocess is asynchronous and then you have to use wait, consider using regular python file autput (with open(filename, 'a') as f: f.write(output)) - this is a synchronous operation, you will not need the wait and it's safer (in the theoretic case that writing will take even more than 1 second for some reason).